src/client/datascience/commands/notebookCommands.ts

```diff
@@ -95,12 +97,15 @@ export class NotebookCommands implements IDisposable {
                   };
         }
         if (options.identity) {
+            const isLocal = isLocalLaunch(this.configService);
             // Make sure we have a connection or we can't get remote kernels.
-            const connection = await this.notebookProvider.connect({
-                getOnly: false,
-                disableUI: false,
-                resource: options.resource
-            });
+            const connection = isLocal
```

Comment:
When swtching kernels, we do'nt need to start or connect the notebook, thats unnecessary. 

Basically its impossible to switch kernels if you have an interpreter without ipykernel or the like. This could explain why users aren unable to start notebooks as well, they cannot change the kernel.